src/services/pegout-data.processor.ts

```diff
@@ -1,27 +1,70 @@
 import {getLogger, Logger} from 'log4js';
+import {inject} from '@loopback/core';
 import {RskTransaction} from '../models/rsk/rsk-transaction.model';
 import {BRIDGE_METHODS, getBridgeSignature} from '../utils/bridge-utils';
 import FilteredBridgeTransactionProcessor from '../services/filtered-bridge-transaction-processor';
 import { BridgeDataFilterModel } from '../models/bridge-data-filter.model';
 import { PegoutStatusDataService } from './pegout-status-data-services/pegout-status-data.service';
+import { PegoutStatusDataModel } from '../models/rsk/pegout-status-data.model';
+import { PegoutStatusRulesService } from './pegout-status/pegout-status-rules-services';
+import {ServicesBindings} from '../dependency-injection-bindings';
 
 export class PegoutDataProcessor implements FilteredBridgeTransactionProcessor {
-  logger: Logger;
+  PegoutStatusRulesService: PegoutStatusRulesService;
   pegoutStatusDataService: PegoutStatusDataService;
+  logger: Logger;
 
-  constructor(pegoutStatusDataService: PegoutStatusDataService) {
+  constructor(
+    @inject(ServicesBindings.PEGOUT_STATUS_SERVICE)
+    PegoutStatusRulesService: PegoutStatusRulesService,
+    pegoutStatusDataService: PegoutStatusDataService)
+ {
     this.logger = getLogger('pegoutDataProcessor');
     this.pegoutStatusDataService = pegoutStatusDataService;
+    this.PegoutStatusRulesService = PegoutStatusRulesService;
   }
   async process(rskTransaction: RskTransaction): Promise<void> {
```

Comment:
This method will receive four kind of transactions:
- pegout request transactions (emptyDataFilter)
- pegout creation transactions (updateCollections)
- pegout confirmed transactions (updateCollections)
- pegout signed transactions (addSignature)

Keep this in mind.

In the way it is done now it looks like it will only process one kind of transaction.